scheduler/core/components/optimizer/greedymax.py

```diff
@@ -5,23 +5,37 @@
 
 from dataclasses import dataclass
 from datetime import datetime, timedelta
-from typing import Dict, FrozenSet, List, Optional, Tuple, Union
+from enum import Enum
+from typing import final, Dict, FrozenSet, List, Optional, Tuple
 
+import matplotlib.pyplot as plt
+import numpy as np
+import numpy.typing as npt
+from lucupy.minimodel import (NIR_INSTRUMENTS, Group, NightIndex, Observation, ObservationClass, ObservationID,
+                              ObservationStatus, Program, QAState, Sequence, Site, UniqueGroupID, Wavelengths)
+from lucupy.minimodel.resource import Resource
+from lucupy.types import Interval, ZeroTime
+
+from scheduler.core.calculations import GroupData, NightTimeSlotScores
 from scheduler.core.calculations.selection import Selection
-from scheduler.core.calculations import GroupData
-from scheduler.core.plans import Plan, Plans
 from scheduler.core.components.optimizer.timeline import Timelines
+from scheduler.core.plans import Plan, Plans
+from scheduler.services import logger_factory
 from .base import BaseOptimizer, MaxGroup
-from . import Interval
 
-from lucupy.minimodel import Program, Group, Observation, Sequence
-from lucupy.minimodel import ObservationID, Site, UniqueGroupID, QAState, ObservationClass, ObservationStatus
-from lucupy.minimodel.resource import Resource
-import numpy as np
-import numpy.typing as npt
-import matplotlib.pyplot as plt
+logger = logger_factory.create_logger(__name__)
+
+
+@final
+class Mode(str, Enum):
+    """
+    TODO: Get rid of this later as per GSCHED-413.
+    """
+    SPECTROSCOPY = 'spectroscopy'
```

Comment:
Pretty sure this will ultimately have to change since Bryan said it is an issue in the `Atom` code with things not being set properly, but for now, this should do.